src/pybind/mgr/dashboard_v2/HACKING.rst

```diff
@@ -508,3 +508,160 @@ The settings management implementation will make sure that if you change a
+will return all executing and finished tasks which namespace starts with
+``rbd/``.
+
+To prevent the finished tasks list from growing unboundly, the finished tasks
+will be maintained in memory for 1 minute. After a minute, when the finished
+task information is retrieved, either by the summary controller or by the task
+controller, it is automatically deleted from the list and it will not be
+included in further task queries.
+
+Each executing task is represented by the following dictionary::
+
+  {
+    'namespace': "namespace",  # str
+    'metadata': { },  # dict
+    'begin_time': 0.0,  # float
+    'progress': 0  # int (percentage)
+  }
+
+Each finished task is represented by the following dictionary::
+
+  {
+    'namespace': "namespace",  # str
+    'metadata': { },  # dict
+    'begin_time': 0.0,  # float
+    'end_time': 0.0,  # float
+    'latency': 0.0,  # float
+    'progress': 0  # int (percentage)
+    'success': True,  # bool
+    'ret_value': None,  # object, populated only if 'success' == True
+    'exception': None,  # str, populated only if 'success' == False
+  }
+
+
+How to updated the execution progress of an asynchronous task?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The asynchronous tasks infrastructure provides support for updating the
+execution progress of an executing task.
+The progress can be updated from within the code the task is executing, which
+usually is the place where we have the progress information available.
+
+To upgrade the progress from within the task code, the ``TaskManager`` class
+provides a method to retrieve the current task object::
+
+  TaskManager.current_task()
+
+The above method returns the current ``AsyncTask`` object. The ``AsyncTask``
+object provides two public methods to update the execution progress value: the
+``set_progress(percentage)``, and the ``inc_progress(delta)`` methods.
+
+The ``set_progress`` method receives as argument an integer value representing
+the absolute percentage that we want to set to the task.
+
+The ``inc_progress`` method receives as argument an integer value representing
+the delta we want to increment to the current execution progress percentage.
+
+Now we show a full example of a controller that triggers a new task and
+updates its progress:
+
+.. code-block:: python
+
+  from __future__ import absolute_import
+  import random
+  import time
+  import cherrypy
+  from ..tools import TaskManager, ApiController, BaseController
+
+
+  @ApiController('dummy_task')
+  class DummyTask(BaseController):
+      def _dummy(self):
+          top = random.randrange(100)
+          for i in range(top):
+              TaskManager.current_task().set_progress(i*100/top)
+              # or TaskManager.current_task().inc_progress(100/top)
+              time.sleep(1)
+          return "finished"
+
+      @cherrypy.expose
+      @cherrypy.tools.json_out()
+      def default(self):
+          task = TaskManager.run("dummy/task", {}, self._dummy)
```

Comment:
Can you be more clear on what you mean by `should be part of the definition of the task`?

The `namespace` parameter is supposed to be a dynamic property on purpose, if the controller developer wants all its tasks to have the same `namespace` string then it can declare a constant variable with that string.

Reply:
The task name is typically very similar to the method name plus class or package name. There is no need to make this a dynamic property, except if you have a clear usecase in mind.

Reply:
@sebastian-philipp If you only have one single `task_name` as the sole identifier of a single task in the queue, how do you expect to handle different tasks performing the same base action if you are not also differentiating them based on their arguments?

You actually gave a pretty good example of how what you are proposing would not work (in another comment), even though you presented it as an argument against the current implementation:

> TaskManager.run('set_pg_count_per_pool', 42, lambda: set_pg_count_per_pool(mypool, 42))
> TaskManager.run('set_pg_count_per_pool', 12, lambda: set_pg_count_per_pool(mypool, 12))
> TaskManager.run('set_pg_count_per_pool', 42, lambda: set_pg_count_per_pool(mypool, 42))

Whereas what is above would be expected to run two different tasks, and have the third call being returned the status of the first call (given it's an idempotent operation), how would you handle such a sequence of operations if you only had one single, static identifier `task_name`?